blog/_posts/2020-02-12-local-dev.md

```diff
@@ -0,0 +1,86 @@
+  - tilt
+  - local
+---
+
+Many teams that use Kubernetes for production don't use Kubernetes in development to avoid Kubernetes pain points (like the slow inner-loops of building container images). But these teams still feel the pains of multi-service development:
+* Updating the right subset of microservices after you edit files is manual/error-prone/ad hoc even if you're just running bare processes on your laptop.
+* Finding a problem across many services? Playing 20 questions is as clumsy with `tail -f` or `ps` as `kubectl`.
+* Onboarding new users to this clusterf...un? You want them to be productive on day 1 without having to learn the whole stack, regardless of where it's running.
+
+Now you can use Tilt to solve these pains of multi-service development even if you aren't using Kubernetes for local development. Tilt can improve workflows built atop `foreman`, `docker-compose`, `edward`, a custom `start.sh, or a wiki page with tmux instructions.
+
+Expanding to support non-Kubernetes development environments gets us closer to our promise: if you use Kubernetes in Prod, you need Tilt for Dev.
+
+## What You Get
+
+After writing a Tiltfile, you'll have a better local development experience:
+* Start your whole app just by running `tilt up`
+* Update microservices automatically as you edit
+* Web UI that lets you see problems across services and zoom in on issues
+* Easily run workflows like resetting database state
+
+![Screenshot of Tilt with Local Dev](/assets/images/local-dev/screenshot.png)
+
+
+From this UI, you can see:
+* The list of servers and workflows ("Resources") in the sidebar on the right
+* All logs, multiplexed together
+* The status of each Resource (e.g. if a server died, its status will be red instead of green)
+* Logs for just one resource, by clicking on it in the sidebar
+
+## local_resource with serve_cmd
+Let's look at a sample Tiltfile for a project (simplified from our sample project [abc123](https://github.com/windmilleng/abc123/blob/master/Tiltfile.local)) that uses Tilt for two different cases:
+* a Go frontend that requires a build on each change
+* a Python backend ("numbers") that watches files and updates itself, except for a change to `requirements.txt`.
+
+```python
+local_resource(
+  'fe',
+  'go install github.com/windmilleng/abc123/fe',
+  serve_cmd='cd fe && fe --port=8000 --backend=localhost:8002',
+  deps=['fe'])
+
+local_resource(
+  'numbers',
+  'cd numbers && pip install -r requirements.txt',
+  serve_cmd='cd numbers && python app.py --port=%d' % (numbers_port),
+  # numbers is a flask app that handles most changes via hot reloading, but
+  # not requirements.txt changes
+  deps=['numbers/requirements.txt'])
+```
+
+This example project has clean separation between services. Tilt can handle more complex setups using options to `local_resource` that you can learn about in the [API reference](https://docs.tilt.dev/api.html#api.local_resource).
+
+## Ensuring consistency with `local_resource`
```

Comment:
At first I read this as "Oh, this feature is consistent with `local_resource`, which was a bit confusing.

Maybe reword as: "Check environments with `local_resource`"? "Check pre-reqs with `local_resource`"?